### DISCUSSION TITLE
[SO2.x] Separate Drive Mount for PCAPs and NSM?

### DISCUSSION THREAD

**Discussion by RagnarSTS** [Security-Onion-Solutions/securityonion#2164] [Unsupported Versions] [answered]
I'm pretty sure I can remount things properly by hand but want to make sure I'm not breaking anything too badly. The installer only gives an option to set the entire NSM on a separate drive. If one doesn't want to burn SSDs on packet capture and uses spinning platters, then this also forces all regular queries to be as slow as the spinning platters. 
I don't generally mind if pulling a PCAP takes a minute or so.  
I do mind if my queries take a minute or so.  

Anybody have a repeatable and recommended method of putting NSM metadata on a faster drive set and putting the actual PCAPs on a big fat slow one?

**Comment by 0nnyx** [comment]
Also interested with an option on default SO to set the /nsm/pcap on HDD and keep the rest on SSD

**Comment by TOoSmOotH** [comment, marked as answer]
This does not preserve any PCAP. You can move it to the new disk if you want to with sym links. This is just from my head and not a complete cut and paste so do your research.

1. mkdir -p /mnt/pcap
2. Add it to your fstab
3. mount -a to make sure it mounts ok
4. mkdir -p /mnt/pcap/{pcap,pcapout,pcapindex,pcaptmp}
5. so-steno-stop
6. mv /nsm/pcap /nsm/pcap.old
7. ln -s /nsm/pcap /mnt/pcap/pcap
8. Do the other PCAP directories the same
9. salt-call state.highstate
10. if everything works delete the .old directories.

**Reply by RagnarSTS** [reply]
And you totally got me to make my link backwards on the first try.  :)

**Comment by RagnarSTS** [comment]
So after doing a `du -sh /var` to check the /nsm folder, I decided that all I really had to move was the elasticsearch info.  
Attempting to follow the above idea of adding a drive, mounting it, and then stopping services and moving things didn't work so good.
Straight up, I couldn't keep kibana and elastic down long enough to move the files. 
Single user mode isn't much helpful because of lvm and lack of mapper. 
I'm now attempting a fresh load, and prior to second stage of setup will copy and link the elastic to a /mnt/elastic.  
Definitely not ideal, nor intuitive. I think we really need a way to specify PCAPs go to a different place.  All things SIEM/NSM are not necessarily PCAP.  PCAPs can be the inevitable growth and need to be relatively easy to not only start out separated out from the rest, but perhaps move there after system is in production.

...yes I know storage system... There is a middle ground between single standalone and full hardware distribution.  :)

**Reply by RagnarSTS** [reply]
Seems functional.  With a fresh load and therefore much less data to move around:
Partitioned odd duck SSD; formatted xfs.  Added noatime to all mounts.
then did something that my or may not have looked like this:
`so-elastalert-stop && so-elasticsearch-stop && so-kibana-stop && mv /nsm/elasticsearch /mnt/elastic/ && ln -s /mnt/elastic/elasticsearch /nsm/elasticsearch`
After a reboot to ensure full system stability and remounting, seems alright and spunky again.  We'll see how it works out over a few weeks.
